src/member/member.controller.ts

```diff
@@ -172,6 +176,32 @@ export class MemberController {
     };
     await this.exportQueue.add(exportJob, { removeOnComplete: true, removeOnFail: true });
   }
+  @Post('saleLeadMemberData')
+  public async getSaleLeadMemberData(
+    @Body() requestDto: SaleLeadMemberDataResquestDTO,
+  ): Promise<SaleLeadMemberDataResponseDTO> {
+    const errors = [];
+
+    if (!requestDto.appId || typeof requestDto.appId !== 'string' || requestDto.appId.trim() === '') {
+      errors.push('appId must be a string and cannot be empty');
+    }
+
+    if (errors.length > 0) {
+      throw new BadRequestException({
+        message: 'Invalid request parameters',
+        errors: errors,
+        statusCode: HttpStatus.BAD_REQUEST,
+      });
+    }
+
+    try {
+      return await this.memberService.getSaleLeadMemberData(requestDto.memberIds, requestDto.appId);
+    } catch (error) {
+      console.error('Error fetching sale lead member data:', error);
+
+      throw new HttpException('Internal Server Error', HttpStatus.INTERNAL_SERVER_ERROR);
```

Comment:
I think it will be fine to use APIException to throw error , this function can return 400 when api get error

Reply:
i have removed HttpStatus / HttpException in the import and use APIException when an error occurs